### DISCUSSION TITLE
Shall the components split into basic and advance packages?

### DISCUSSION THREAD

**Discussion by cairabbit** [dotnetcore/BootstrapBlazor#3625] [Q&A]
This amazing component library now becomes too heavy. at least to me. Some exellent features are added to the compoents, that could be wonderful addon to some people but may be trouble to others. For example, the Layout compoent has different layout for user authenticated, unauthenticated, not found. which is far beyond the definition of a Layout component. The design makes it fit to some specific scenario therefor also making it not suitable for other scenario. Which sometimes is quite annoying as unneeded features stopping you catering some basic needs. 

I just wonder if we could split (or redesign) the components into basic components and advance components, where the basic components only concert about the UI and UI interactions. No other logic involved even authentication. All these basic components could be the same as what bootstrap have, or a few more basic components, say BootstrapBlazor. Then all other complex components like table, tab documents etc could be in another package, say BootstrapBlazor.Advance. In this way, if I can use BootstrapBlzor just for UI and UI interaction. Any auth/logic involved components should be separated. for example, <AuthenticatedView> requires authentication involved. Or even more specific, BootstrapBlazor.Advance.TabDocument, BootstrapBlazor.Advance.DataTable (include search, CURD) etc.

Does anyone agree?

Thanks.

**Comment by ArgoZhang** [comment]
First, splitting the component library itself is a very destructive update, which will cause old users and old projects not to work properly.

Among all the component libraries currently, our component library is not large (file size), but there are more components. We support three target frameworks and the file size is only 2.45M

I have always had an idea to split the component library

- BootstrapBlazor.Core
- BootstrapBlazor.Components

I gave up this idea because of the upgrade issue.
